src/queries/query.ts

```diff
@@ -28,26 +37,86 @@ export interface BasicQueryData {
 export type QueryExecutor<Query extends BaseQueryFormat, Result extends BaseQueryResult> = (data: BasicQueryData, query: readonly Query[]) => Result;
 
 type SupportedQueries = {
-	[QueryType in Query['type']]: QueryExecutor<QueryArgumentsWithType<QueryType>, BaseQueryResult>
+	[QueryType in Query['type']]: SupportedQuery<QueryType>
+}
+
+interface SupportedQuery<QueryType extends BaseQueryFormat['type']> {
+	executor:        QueryExecutor<QueryArgumentsWithType<QueryType>, BaseQueryResult>
+	asciiSummarizer: (formatter: OutputFormatter, processed: PipelineOutput<typeof DEFAULT_DATAFLOW_PIPELINE>, queryResults: BaseQueryResult, resultStrings: string[]) => boolean
+	schema:          Joi.ObjectSchema
 }
 
 export const SupportedQueries = {
-	'call-context':   executeCallContextQueries,
-	'dataflow':       executeDataflowQuery,
-	'id-map':         executeIdMapQuery,
-	'normalized-ast': executeNormalizedAstQuery
+	'call-context': {
```

Comment:
ok this is no half the way i wanted it to be :sweat_smile: could we move these definitions away from here so that the overview is nice and readable? 
For example:

`'call-context': CallContextQueryDefinition`